js/shop.js

```diff
@@ -1,102 +1,26 @@
-
-/*
- get the course number for the canvas course
+/**
+ * allowed_terms is a white list of Canvas enrollment_term_ids where shopping is allowed
+ * for a specific school
+ * NOTE - the term ids in allowed_terms must be strings, not ints
+ * @type {string[]}
  */
-function get_course_number() {
-  var page_url = window.location.pathname;
-  var pat = /\/courses\/(\d+)/g;
-  var match = pat.exec(page_url);
-  if (match) {
-    course_id = match[1];
-    return course_id;
-  }
-  return 0;
-}
-
-function get_sis_user_id(canvas_user_api_data) {
-  var user_id = null;
-  if (canvas_user_api_data) {
-    if (canvas_user_api_data['sis_user_id'] && canvas_user_api_data['sis_user_id'].trim()) {
-      user_id = canvas_user_api_data['sis_user_id'].trim();
-    } else if (canvas_user_api_data['login_id'] && canvas_user_api_data['login_id'].trim()) {
-      user_id = canvas_user_api_data['login_id'].trim();
-    }
-  }
-  return user_id;
-}
-
-function is_course_available(course_workflow) {
-  return course_workflow.localeCompare('available') == 0;
-}
-
-function term_allowed(term_id) {
-  return jQuery.inArray(term_id, allowed_terms) > -1;
-}
-
-var current_user_id = ENV['current_user_id'];							// the canvas id if the current user
-var user_url = '/api/v1/users/' + current_user_id + '/profile';			// the url to the user profile api call
-
-var course_id = get_course_number();
-var course_url = '/api/v1/courses/' + course_id ;
-
-var data_tooltip = 'More info about access during shopping period';
-var shopping_help_doc_url = 'https://wiki.harvard.edu/confluence/display/canvas/Course+Shopping';
-var tooltip_link = '<a data-tooltip title="' + data_tooltip + '" target="_blank" href="' + shopping_help_doc_url + '"><i class="icon-question"></i></a>';
-var login_url = window.location.origin+"/login";
-var no_user_canvas_login = "<div class='tltmsg tltmsg-shop'><p class='participate-text'>Students: <a href=\""+login_url+"\">login</a> to get more access during shopping period." + tooltip_link + "</p></div>";
-
-var on_admin_page = ((window.location.pathname).indexOf('settings') != -1);
-var on_speed_grader_page = ((window.location.pathname).indexOf('speed_grader') != -1);
-var on_submissions_page = ((window.location.pathname).indexOf('submissions') != -1);
-var on_special_page = on_admin_page || on_speed_grader_page || on_submissions_page;
-
-var user_enrolled = false;
-var is_shopper = false;
-var is_teacher = false;
-var is_student = false;
-
-var shopping_banner = jQuery('<div/>', {
-  id: 'course-shopping',
-  class: 'tltmsg'
-});
-
-/*
- check to see if the '#unauthorized_message' is being rendered  and only proceed
- with additional checks to show shopping messages if authorized
- */
-var authorized = $('#unauthorized_message').length > 0 ? false : true;
-
-if (authorized){
+var allowed_terms = ['3', '4', '179', '595', '596', '603', '487', '569', '1','8','447','591','597', '342'];
 
+if (unauthorized_message_not_shown){
```

Comment:
As commented in the other PR, any particular reason to switch to unauthorized_message_not_shown?